apps/hash-frontend/src/shared/notifications-context.tsx

```diff
@@ -115,64 +120,98 @@ export const NotificationsContextProvider: FunctionComponent<
+            { parameter: authenticatedUser?.accountId },
+          ],
+        },
+        generateVersionedUrlMatchingFilter(
+          types.entityType.notification.entityTypeId,
+          { ignoreParents: false },
+        ),
+        {
+          any: [
             {
               equal: [
-                { path: ["ownedById"] },
-                { parameter: authenticatedUser?.accountId },
-              ],
-            },
-            generateVersionedUrlMatchingFilter(
-              types.entityType.notification.entityTypeId,
-              { ignoreParents: false },
-            ),
-            {
-              any: [
                 {
-                  equal: [
-                    {
-                      path: [
-                        "properties",
-                        extractBaseUrl(
-                          types.propertyType.archived.propertyTypeId,
-                        ),
-                      ],
-                    },
-                    // @ts-expect-error -- We need to update the type definition of `EntityStructuralQuery` to allow for this
-                    //   @see https://linear.app/hash/issue/H-1207
-                    null,
+                  path: [
+                    "properties",
+                    extractBaseUrl(types.propertyType.archived.propertyTypeId),
                   ],
                 },
+                // @ts-expect-error -- We need to update the type definition of `EntityStructuralQuery` to allow for this
+                //   @see https://linear.app/hash/issue/H-1207
+                null,
+              ],
+            },
+            {
+              equal: [
                 {
-                  equal: [
-                    {
-                      path: [
-                        "properties",
-                        extractBaseUrl(
-                          types.propertyType.archived.propertyTypeId,
-                        ),
-                      ],
-                    },
-                    { parameter: false },
+                  path: [
+                    "properties",
+                    extractBaseUrl(types.propertyType.archived.propertyTypeId),
                   ],
                 },
+                { parameter: false },
               ],
             },
           ],
         },
+      ],
+    }),
+    [authenticatedUser],
+  );
+
+  const {
+    data: notificationRevisionsData,
+    loading: loadingNotificationRevisions,
+    refetch: refetchNotificationRevisions,
+  } = useQuery<
+    StructuralQueryEntitiesQuery,
+    StructuralQueryEntitiesQueryVariables
+  >(structuralQueryEntitiesQuery, {
+    pollInterval: 5_000,
+    variables: {
+      includePermissions: false,
+      query: {
+        filter: getNotificationsQueryFilter,
+        graphResolveDepths: zeroedGraphResolveDepths,
+        /**
+         * We need to obtain all revisions of the notifications
+         * to determine when they were created.
+         */
```

Comment:
Can you please add a comment to the effect of 'Update this when H-1098 is implemented' so we know where to come and change when we have `createdAt` natively. And in the other places (quick notes?) that you can remember while we're at it

Reply:
Have added a todo which links the task in https://github.com/hashintel/hash/pull/3511/commits/e1434afef3d6e6f5580e92e5bb923bb0c936be96